### ISSUE TITLE
[vault-env] Fetch image descriptor ignore registry mirror.

### ISSUE THREAD

**Issue by fdpeiter** [banzaicloud/bank-vaults#1398]
> Bugs should be filed for issues encountered whilst operating bank-vaults.
> You should first attempt to resolve your issues through the community support
> channels, e.g. Slack, in order to rule out individual configuration errors.
> Please provide as much detail as possible.

**Describe the bug**:
After DockerHub implemented it's rate limit, we've pushed via Kops a configuration to add a Registry Mirror to the nodes. Although this worked fine for pods that doesn't use vault secrets, whenever a pod need mutation, the configuration seems to be ignored:

```
time="2021-08-13T19:36:35Z" level=error msg="admission webhook error: cannot fetch image descriptor: GET https://index.docker.io/v2/tfgco/fluxcloud/manifests/v0.5.0: TOOMANYREQUESTS: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit" app=vault-secrets-webhook
```

Checking the documentation, we've noticed two environment variables that should help with this, enabling us to inject a DockerHub credential using a Kubernetes Secret: https://github.com/banzaicloud/bank-vaults/blob/master/charts/vault-secrets-webhook/values.yaml#L59

We've followed the https://banzaicloud.com/docs/bank-vaults/mutating-webhook/configuration/#registry-access tutorial, creating a secret with the DockerHub credentials:

```bash
kubectl -n vault-secrets-webhook create secret docker-registry dockerhub --docker-username=${DOCKER_USERNAME} --docker-password=$DOCKER_PASSWORD
```

We've then proceeded to define both env variables:
```yaml
        - name: DEFAULT_IMAGE_PULL_SECRET
          value: dockerhub
        - name: DEFAULT_IMAGE_PULL_SECRET_NAMESPACE
          value: vault-secrets-webhook
```
The errors still persisted and checking the code, we don't believe one of the conditions for the variables to be used will ever be true:  https://github.com/banzaicloud/bank-vaults/blob/b38c0a7d868a75f549a1307a585d425bbe1e4c06/pkg/webhook/registry.go#L108. It seems that the `containerInfo.Namespace` will never be empty.

**Expected behaviour**:
After configuring the DockerHub credentials and both default values for the pull secrets, we've expected pods that don't have the ImagePullSecrets configuration  used the default ones to authenticate in DockerHub.

**Steps to reproduce the bug**:
1. Reach DockerHub Anonymous rate limit (100 pulls/6hrs).
2. Configure Docker Hub credentials in the same Namespace as VSW.
3. Update VSW Deployment adding DEFAULT_IMAGE_PULL_SECRET = dockerhub and  DEFAULT_IMAGE_PULL_SECRET_NAMESPACE = vault-secrets-webhook.
4. VSW used the Default values whenever a pod spec didn't had ImagePullSecrets configured.

**Additional context**:
The problem happened in multiple clusters, with different K8s versions.

**Environment details**:
- Kubernetes version (e.g. v1.10.2): 1.18.18/1.20.1
- Cloud-provider/provisioner (e.g. AKS, GKE, EKS, PKE etc): AWS
- bank-vaults version (e.g. 0.4.17): 1.12.0
- Install method (e.g. helm or static manifests): helm
- Logs from the misbehaving component (and any other relevant logs): 
- Resource definition (possibly in YAML format) that caused the issue, without sensitive data:
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app.kubernetes.io/component: mutating-webhook
    app.kubernetes.io/instance: vaultsecretswebhook
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: vault-secrets-webhook
    helm.sh/chart: vault-secrets-webhook-1.6.3-4
  name: vaultsecretswebhook-vault-secrets-webhook
  namespace: vault-secrets-webhook
  resourceVersion: "110195949"
  selfLink: /apis/apps/v1/namespaces/vault-secrets-webhook/deployments/vaultsecretswebhook-vault-secrets-webhook
  uid: bef97215-911e-488a-b380-d3cfbeac96b9
spec:
  progressDeadlineSeconds: 600
  replicas: 3
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app.kubernetes.io/instance: vaultsecretswebhook
      app.kubernetes.io/name: vault-secrets-webhook
  strategy:
    rollingUpdate:
      maxSurge: 25%
      maxUnavailable: 25%
    type: RollingUpdate
  template:
    metadata:
      creationTimestamp: null
      labels:
        app.kubernetes.io/instance: vaultsecretswebhook
        app.kubernetes.io/name: vault-secrets-webhook
        security.banzaicloud.io/mutate: skip
    spec:
      containers:
      - env:
        - name: TLS_CERT_FILE
          value: /var/serving-cert/tls.crt
        - name: TLS_PRIVATE_KEY_FILE
          value: /var/serving-cert/tls.key
        - name: LISTEN_ADDRESS
          value: :8443
        - name: DEFAULT_IMAGE_PULL_SECRET
          value: dockerhub
        - name: DEFAULT_IMAGE_PULL_SECRET_NAMESPACE
          value: vault-secrets-webhook
        - name: VAULT_ENV_IMAGE
          value: ghcr.io/banzaicloud/vault-env:1.12.0
        - name: VAULT_ADDR
          value: https://xxx
        - name: VAULT_IMAGE
          value: vault:1.6.2
        - name: VAULT_PATH
          value: xxx
        - name: VAULT_SKIP_VERIFY
          value: "true"
        image: banzaicloud/vault-secrets-webhook:1.12.0
        imagePullPolicy: Always
        name: vault-secrets-webhook
        ports:
        - containerPort: 8443
          protocol: TCP
        readinessProbe:
          failureThreshold: 3
          httpGet:
            path: /healthz
            port: 8443
            scheme: HTTPS
          periodSeconds: 10
          successThreshold: 1
          timeoutSeconds: 1
        resources: {}
        securityContext:
          allowPrivilegeEscalation: false
          runAsUser: 65534
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /var/serving-cert
          name: serving-cert
      dnsPolicy: ClusterFirst
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      serviceAccount: vaultsecretswebhook-vault-secrets-webhook
      serviceAccountName: vaultsecretswebhook-vault-secrets-webhook
      terminationGracePeriodSeconds: 30
      volumes:
      - name: serving-cert
        secret:
          defaultMode: 420
          secretName: vaultsecretswebhook-vault-secrets-webhook-webhook-tls
```

Secret:
```
apiVersion: v1
data:
  .dockerconfigjson: <base64secret>
kind: Secret
```
/kind bug

**Comment by thatarchguy** [comment]
We're running into a similar problem. Everything works fine with public images. When we specify our registry mirror we run into problems. We have tried to use those env vars to reference our docker config secret with no avail. We even tried specifying ImagePullSecrets directly in the pod spec we're deploying and it's still having trouble. 
  
 My error is slightly different:
  ```
  time="2021-08-18T19:57:33Z" level=error msg="Admission review request failed" app=vault-secrets-webhook dry-run=false error="cannot fetch image descriptor: unrecognized challenge: " kind=v1/Pod name= ns=foobar op=create path=/pods request-id=a07ee32c-f21a-4ca5-bd21-bf899635dace webhook-id=vault-secrets-pods webhook-kind=mutating wh-version=v1beta1
```
    
    Kubernetes version (e.g. v1.10.2): v1.19.12
    Cloud-provider/provisioner (e.g. AKS, GKE, EKS, PKE etc): Rancher on AWS EC2
    bank-vaults version (e.g. 0.4.17): 1.13.1
    Install method (e.g. helm or static manifests): helm


A workaround we found is to specify the `command` field in your pod spec. 
```
command: ["/bin/run-my-thing"]
```

This is because the webhook only pulls the image and inspects it to get the command if one is not specified in the spec. 
https://github.com/banzaicloud/bank-vaults/blob/v1.13.1/pkg/webhook/pod.go#L247
```go
		if len(args) == 0 {
			imageConfig, err := mw.registry.GetImageConfig(ctx, mw.k8sClient, ns, &container, podSpec) // nolint:gosec
			if err != nil {
				return false, err
			}
```

**Comment by bonifaido** [comment]
This has been fixed in #1409. Please try the most recent version and feel free to open up if the issue still persists.
